File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        if not s:
            return 0
        i=j=0
        max_len=1
        hashmap={}

        def get_max_freq():
            count=int(max(hashmap.values()))
            return count

        while(j<len(s)):
            if s[j] in hashmap:
                hashmap[s[j]]=hashmap[s[j]]+1
            else:
                hashmap[s[j]]=1

            max_freq=get_max_freq()
            if ((j-i)+1)-max_freq<=k:
                max_len=max(max_len,((j-i)+1))
                j+=1
            else:
                hashmap[s[i]]=hashmap[s[i]]-1
                i+=1
                j+=1
        return max_len

        

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        var freqMap=new Dictionary<char,int>();
        int start=0;
        int end=0;
        int maxLength=0;
        int mostFreqCharCount=0;
        while(end<s.Length){
            if(freqMap.ContainsKey(s[end])){
                freqMap[s[end]]++;
            }else{
                freqMap[s[end]]=1;
            }
            mostFreqCharCount=Math.Max(mostFreqCharCount,freqMap[s[end]]);
            // var mostFreqCharCount=freqMap.OrderByDescending(kvp=>kvp.Value).Select(kvp=>kvp.Value).First();
            if((end-start+1)-mostFreqCharCount>k){
                freqMap[s[start]]--;
                start++;
            }
            maxLength=Math.Max(maxLength,(end-start+1));
            end++;
        }
        return maxLength; 
    }
}